### PR TITLE
document with useful code snippets

### PR DESCRIPTION
**Pull request by Th3SnowyOwl**
I am suggesting a .sh script to make it easier for gitpod users to set up or compile for mod creation. Anyone have any suggestions on what is could put in this to make it better? You could just put this in the docs as something like "moddev.sh"
[moddev.sh.txt](https://github.com/ninjamuffin99/Funkin/files/6345302/moddev.sh.txt)

### CODE REVIEW

**Review by actuallyasmartname — APPROVED** [review]

File: moddev.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+sudo apt update && sudo apt upgrade -y
+
+#Haxe
+sudo add-apt-repository ppa:haxe/releases -y && sudo apt install haxe -y && mkdir ~/haxelib && haxelib setup ~/haxelib
+
+#MoreHaxe (Flixel)
+haxelib setup && haxelib install lime && haxelib install flixel && haxelib install openfl && haxelib run lime setup -y
+
+#Setup
+haxelib install flixel && haxelib install flixel-addons && haxelib install flixel-ui && haxelib install hscript && haxelib install newgrounds
```
[the remark this answers is not in the document]
Comment (Th3SnowyOwl) [reply moddev.sh]: yeah im still fixing some things I just updated it

**Comment by Th3SnowyOwl** [comment]
Okay, the script should be good now.

File: moddev.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (Th3SnowyOwl) [reply moddev.sh]: i didn't see that in the readme lol

**Comment by OPNA2608** [comment]
What exactly is this? A shell script one is supposed to execute on their machine? A collection of useful code snippets?

**Comment by Th3SnowyOwl** [comment]
ignore note_assets, that's going in another post, but (OPNA Readme: Gitpod streamlines developer workflows by providing prebuilt, collaborative development environments in your browser - powered by VS Code.)Description?. If you would like to look at the webpage, what it allows you to do is instantly clone and edit a GitHub repository on your browser and give you a Linux terminal to run commands on for compiling and editing. [https://www.gitpod.io/about](https://www.gitpod.io/about).

**Comment by Th3SnowyOwl** [comment]
Sorry for not elaborating more, this is supposed to be a useful shell script you can run in GitPod to install all the necessary libraries.

File: assets/shared/images/NOTE_assets.xml [hunk 2]
```diff
@@ -6,7 +6,6 @@
 	<SubTexture name="arrowLEFT0000" x="310" y="235" width="153" height="157"/>
 	<SubTexture name="arrowRIGHT0000" x="157" y="235" width="153" height="157"/>
 	<SubTexture name="arrowUP0000" x="784" y="232" width="157" height="153"/>
-	<SubTexture name="blue0000" x="1850" y="154" width="157" height="154"/>
```
Comment (OPNA2608) [inline assets/shared/images/NOTE_assets.xml:9]: Unrelated to the PR's purpose, please revert this.

**Comment by OPNA2608** [comment]
If the file is supposed to be run by a user, then having the one-liner variant + long variant in there is pointless. It'll run the one-liner, then execute everything below it as well. (It's also missing a shebang)

All the comments about how to fix the APIStuff thing and where to find the compilation instructions are lost on everyone who just executes the script. You should test if the `APIStuff.hx` file exists and create it with the dummy contents if it doesn't. Or at the very least `echo` some details so the user knows they have something else to do after running the script.

Are you *sure* you don't want to write this as a

> *document with useful code snippets*

in Markdown format instead? That seems much more compatible with the one-liner vs multiline, the comments about the APIStuff and compile instructions, and the **`/docs`** folder.

> # Setup instructions
> 
> ## Platform-specific dependencies
> 
> ##### Ubuntu
> 
> ```
> #One liner:
> …
> ```
> 
> ## Common
> 
> ##### APIStuff
> 
> blabla
> 
> ##### Compiling & Running
> 
> blablub

**Comment by Th3SnowyOwl** [comment]
OPNA yes, im bad @ sh, this is really supposed to be more like a txt file, and im' already removing the note assets, it was there by accident.

**Comment by Th3SnowyOwl** [comment]
Yeah what you put here is great.

File: assets/shared/images/NOTE_assets.xml (hunk 2 above)
Comment (Th3SnowyOwl) [reply assets/shared/images/NOTE_assets.xml:9]: I don't know how to do that tbh.

**Comment by Th3SnowyOwl** [comment]
Thank yu opna

File: assets/shared/images/NOTE_assets.xml (hunk 2 above)
Comment (Th3SnowyOwl) [reply assets/shared/images/NOTE_assets.xml:9]: just realized i need github desktop lol.

Comment (Steviegt6) [reply assets/shared/images/NOTE_assets.xml:9]: No... you just need Git.

**Pull request closed without merging**